### DISCUSSION TITLE
Is AIO patching mandatory ?

### DISCUSSION THREAD

**Discussion by Thini6588** [Mr-MIBonk/M.I.B._More-Incredible-Bash#207] [Q&A] [answered]
Hi
I am new to MIB and went with strong interest into the pages of the site. Amazing and great job! Gongrats!.
I own an Audi A6 C7.5 and I would like to use MIB just to add the FEC code allowing to update the maps without restrictions.
My MMI firmware is a stock one: MHI2_ER_AU57x_S2519_1, MU 0389.
I have no interest at this stage in applying the ALL-IN-ONE Patch, because no need so far for AA and/or CP.
When MIB installed, and back-up done, can I go straight to /add_feccontainer.fec to add the 023000ee only? Or am I obliged to apply ALL-IN-ONE Patch?
In case I change my mind, I have seen that there are no AIO patch available for a …_S2519_1 (only for …_S2519) Would this “…_S2519” patch work in my case?
In case it helps, I propose you to upload my “…_S2519_1” back-up (when done) for further studies.
Thanks in advance for your feedback.
Thinig

**Comment by harman-f** [comment, marked as answer]
Units needs patch to accept new FEC.
However, M.I.B does not have a patch for your FW version.

**Reply by Bossweb2K** [reply]
My unit 5F has the FW, MHI2_ER_AU57x_K3663 (MU1425)

A couple of years ago, the unit was patched but the map FECs (0230002f) does not allow me to update the latest ones available.

How can I use the MIB Tool to enter the FEC key to have lifetime updates (023000EE)?

**Reply by harman-f** [reply]
in that case addFec should be enough

**Reply by Bossweb2K** [reply]
With the fec keys that I currently have, I couldn't update the maps last year or this one.

To add the fec key **(023000EE)**, you would have to enter this menu: 
https://user-images.githubusercontent.com/98130152/174430800-49d49d92-f20d-4bb9-b427-539a0f906cbc.png

 "**add_feccontainer.fec**" and then click "**Add New Fecs to FecContainer.fec**"? 
https://user-images.githubusercontent.com/98130152/174430827-c9694180-4090-4a0e-8edd-fd1302954ffa.png

**Reply by harman-f** [reply]
Yes

**Reply by Bossweb2K** [reply]
OK, thanks for the answer!

I was thinking of installing the AIO 3663_1, but if running from the MIB Tool, the option "**Add New Fecs to FecContainer.fec**", I add the FEC Key **(023000EE)**,... I will do this option 😉

**Reply by Bossweb2K** [reply]
I already did it like this this Sunday, and it worked perfectly! 👌🏻 ... So I was able to update the 2023/2024 maps.

Thanks for the help!! 👏🏻

**Reply by Falconer999** [reply]
I have done this so many times, but only the ones that I already had showed up in the result. Don't care about the other ones but I have 0230024 but never 023000EE

**Comment by Thini6588** [comment]
Hi
Thanks for the feedback.
Would it help if I install MIB just to backup my FW and log in order to upload it for your usage and further developments?

**Reply by harman-f** [reply]
your FW is well known

**Reply by Thini6588** [reply]
OK. Understood. So the absence of patch is more a technical limitation concerning this FW

**Reply by Thini6588** [reply]
Is there a chance to see a patch for my current FW in the future?

**Reply by harman-f** [reply]
Just update to latest 2xxx FW

**Comment by Thini6588** [comment]
Thanks a lot! I'll investigate further this solution and try to find the related procedure.

**Comment by Thini6588** [comment]
Hi. Here is a summary of what I think I have to do:
1- Update to K2589 FW (found in the supporting document section)
2- Install it with the REM, with preliminary activation of the "user defined mode" in SWDL of the GEM (I have Bose system and I would like to make sure no update is made in it)
3- Install MIB and use the "Add Fec" option for the Maps lifetime update
As I am really  new to these processes, I would kindly like your comments and additional advices/tutorials.
Additionally, would I be able to do later-on the AIO Patching process? (in case I finally choose to have AA as well)
Again, thanks for your patience and hints.

**Reply by harman-f** [reply]
3. use M.I.B --> patch unit --> AIO patch

**Reply by Thini6588** [reply]
Thanks a lot. This is appreciated

**Reply by pragakvk** [reply]
Hi @Thini6588 ,
I have exactly the same requirement.
Looking to add maps FEC and have Bose speakers.

May I know if you managed to try and apply the above steps?
How did it go.. please share some steps and tutorials. Thanks

**Reply by Thini6588** [reply]
Hi @pragakvk 
Too busy at the moment and not tried the above steps yet.
I should be able to make it soon and will advise.

**Reply by Thini6588** [reply]
Hi!
I have performed the whole process and I am now able to debrief. It went smoothly until the very final step, when a problem occured unfortunately.
The update to FW K2589 has been performed a week ago. I had activated before the GEM thanks to OBD11 and allow the SWDL mode (in order to be able to select "user defined" installation and to be sure no update is performed in the Bose system). It was not needed as there was no update planned for the Bose system anyway. The update worked perfectly and I drove with the car during one week and found no problem with the MMI (Bose ok, etc).
Earlier today, I have proceed with MIB installation.
It went OK until after the second reboot. Then, I had problem with the writing of the "final script". It stayed at 0% for quite a long time, and then an error message appeared mentioning the error code 113. I have then selected "retry" and it went OK then. I have been able to follow the last steps of the procedure.
When entering the GEM again, I had the bad surprise to see that MIB has not been installed. There was just the usual "production" menu. I realised as well that my GEM was still a 4.0 version.
I think this is the origin of the problem.
I thought (wrongly) that upgrading the FW from S2519_1 to K2589 was also allowing the GEM to be upgraded from 4.0 to 4.1x...

Am I really stuck with this GEM 4.0 version or is there an easy way to upgrade the GEM and retry then the MIB installation?

Thnks for your consideration.
Thini6588

**Comment by rynoc1** [comment]
New here and excited to get my maps updated on my 2018 Audi A5. 

I’ve installed M.I.B, run the AIO Patch and I believe the lifetime FEC is installed. When I moved the maps onto the root of the SD card, I put the entire contents of the M.I.B folder into a folder. Now when I enter GEM, I only see production. Is this normal? Does an SD card always need to be inserted with the M.I.B files on the root in order for MIB to be visible on the GEM?

If so, do you just drag the mib1 and mib2 folders containing the map updates to the root also? Or do these maps folders need putting somewhere else or extracting?

Finally, to update the maps, do we just enter red engineering menu and choose SW update?

Thanks

Ryan

**Reply by harman-f** [reply]
Yes, there has to be a SD card with M.I.B files on it in unit to make it run ;)

Never mix M.I.B and Maps/FW update on same SD

Have individual SDs. There is no need to have M.I.B SD in unit during MAP7FW updates.

Best use REM for updates.

**Reply by rynoc1** [reply]
Thanks for the quick reply! 
I’ll order a second SD card. Is there a limit to the capacity of the SD the car will see? I have a spare 256GB one already, or should I order a small one, say 8 / 16GB. Maybe I could also put some movies on the same SD as the maps

**Reply by harman-f** [reply]
only tried up to 64GB so far. Good question ;)

8GB is fine for M.I.B - FAT32
64GB needed for EU maps - USB stick will also work NTFS/extFAT possible

**Comment by Thini6588** [comment]
> Hi! I have performed the whole process and I am now able to debrief. It went smoothly until the very final step, when a problem occured unfortunately. The update to FW K2589 has been performed a week ago. I had activated before the GEM thanks to OBD11 and allow the SWDL mode (in order to be able to select "user defined" installation and to be sure no update is performed in the Bose system). It was not needed as there was no update planned for the Bose system anyway. The update worked perfectly and I drove with the car during one week and found no problem with the MMI (Bose ok, etc). Earlier today, I have proceed with MIB installation. It went OK until after the second reboot. Then, I had problem with the writing of the "final script". It stayed at 0% for quite a long time, and then an error message appeared mentioning the error code 113. I have then selected "retry" and it went OK then. I have been able to follow the last steps of the procedure. When entering the GEM again, I had the bad surprise to see that MIB has not been installed. There was just the usual "production" menu. I realised as well that my GEM was still a 4.0 version. I think this is the origin of the problem. I thought (wrongly) that upgrading the FW from S2519_1 to K2589 was also allowing the GEM to be upgraded from 4.0 to 4.1x...
> 
> Am I really stuck with this GEM 4.0 version or is there an easy way to upgrade the GEM and retry then the MIB installation?
> 
> Thnks for your consid

**Reply by harman-f** [reply]
did you try to run the update again?

**Reply by Thini6588** [reply]
Do you mean the FW K2xxx update? If correct, no, I did not try it as everything appeared correct during the process.
If MIB update, no. I have just selected "retry" after the 113 error code that showed up after the impossibility to write the "final script" between 2d and 3rd reboot. This "retry" action went very fast and I am not sure the final script has been written (unless it went from 0 to 100% in one second or less...) My problem is possibly that I am still running a 4.0 version of the GEM.

**Reply by harman-f** [reply]
Get a fresh SD card and copy M.I.B to it again.
Run the M.I.B install again.

Looks like the SD or some files were corrupted on your last try.

**Reply by Thini6588** [reply]
OK - Thanks - I'll try asap and let you know. One more question: Is it normal at this stage to have a GEM 4.0? I understand this is a no go to install M.I.B..

**Reply by harman-f** [reply]
without finalscript, GEM is not updated.

**Reply by Thini6588** [reply]
OK - Understood - I'll follow your kind advices in the coming days and keep you informed on the results.
Thanks again for your great and efficient support!

**Reply by Thini6588** [reply]
Hi!
I have tried again and the installation of M.I.B went OK this time. I have realized i was not proceeding the right way before: I have done originally the whole process (update of the FW, and M.I.B. installation, with the car powered ("Car ignition" button "on" but without engine running...) It went OK for the FW update but not for the M.I.B. installation. I have retried the M.I.B installation today, this time with just the "power on" button pushed (to activate the MMI, screen extract, etc). By doing this way, the M.I.B. installation went OK. The M.I.B was activated in the GEM. Perfect!
Then I have been able to to do my back-up, and the AIO patch.
All the fec code have been installed (including 23...ee) and I have been able to install the recent 2023 2024 maps. Perfect!
Unfortunately, the Car Play/Android auto logo does not appear in the MMI menu. I went again into M.I.B. to find out that Carplay/Android auto has been well activated, and my two USB ports as well. In synthesis, everything looks fine and activated when I check into M.I.B. but when I use the MMI, no Carplay/AA logo appears in the main menu!
Do you see where the problem is coming from?
Perhaps I should have done the FW update at the beginning with the just the "power on" and not the "car ignition" button on???
Feel free to let me know your thoughts, and always valuable advices!

**Reply by harman-f** [reply]
Icons only show up if a CP/AA phone is connected and the correct USB port is installed in your car. 
  
Which symbol do you see next to your usb port? Share a picture. 

Always try different cables to connect the phone.

**Reply by Thini6588** [reply]
Here is the picture 
![IMG20230802170103](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/assets/132510234/054a4796-d2ac-4b0a-bfb1-4d7f6bc28094)

**Reply by Thini6588** [reply]
I have tried several cables in both ports and no Android Auto activated in the MMI unfortunately

**Reply by harman-f** [reply]
The one with the phone symbol is the right one. 
the usb. Able I see are not stock CP/AA. Do I see a 30pin iPod cable?!

have you already shared the M.I.B Log file?

**Reply by Thini6588** [reply]
Well noted. The cables you see are one USB/USB C and, correct, one USB/ipod to run my good old iPod Classic 160gb! These are cables coming from my car manufacturer.
I'll by a specific AA cable then to check as well with this one.
A try with my original Android phone cable did not worked either.
I have just uploaded my "logs.zip" file.

**Reply by Thini6588** [reply]
Hi. Are you sure the "Axxx Smartphone Interface" icon should appear in the MMI menu when a phone is connected only?  I am still a bit confused not to see such icon after the patch AIO has been applied. Perhaps the analysis of my logs will tell you more on the problem encountered

**Reply by Thini6588** [reply]
Hi. I have tried multiple cables over the past weeks (including some specific Android Auto that I bought) and none of them makes the "Axxx Smartphone" icon appear.
Hope you can still provide me with some advices on the way to solve the issue.
